pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
```

Comment:
@seh Thanks for the catch on the 0 length value. I have updated to return nil in that case.

Your interesting solution does not preserve the order of the path elements however ("c.a.b" comes back as map[a:map[b:map[c:map[]]]] ). I played with your approach some but didn't get a solution but please let me know if you come up with another elegant one.